unblob/processing.py

```diff
@@ -187,23 +187,8 @@ def _process_chunks(
         self._calculate_entropies(carved_unknown_paths)
 
         for chunk in outer_chunks:
-            carved_valid_path = carve_valid_chunk(extract_dir, file, chunk)
-
-            if chunk.is_encrypted:
-                logger.warning(
-                    "Do not attempt to extract encrypted file",
-                    path=carved_valid_path,
-                    chunk=chunk,
-                )
-                continue
-
-            inpath, outdir = get_extract_paths(extract_dir, carved_valid_path)
-
-            extractor = chunk.handler.EXTRACTOR
-            if extractor is not None:
-                extracted = self._extract_chunk(inpath, outdir, extractor)
-                if extracted and not self.parameters.keep_extracted_chunks:
-                    inpath.unlink()
+            carved_path = carve_valid_chunk(extract_dir, file, chunk)
+            self._extract_chunk(extract_dir, carved_path, chunk)
```

Comment:
I'd put unlinking the input file around these lines so that the lifteteime of the input file is easier to track: it gets created here and gets removed here as well.

Reply:
I have previously thought about moving the call `carve_valid_chunk` inside `_extract_chunk` instead, then the lifetime would be trackable there (it is carved there, removed there). I definitely do not want to complicate `_process_chunks` with file removal logic - it is not that abstraction level: it is about and managing the extraction directory and extracting all of the chunks.